app/assets/javascripts/services/event_notifications_service.js

```diff
@@ -129,6 +129,8 @@ function eventNotifications($timeout, API) {
       type: levelToType(type),
       message: message,
       data: notificationData,
+      actionTitle: __('View details'),
+      actionCallback: this.viewDetails,
```

Comment:
We do not want all notifications to have the `View Details` link.

For e.g. in v2v, only the unsuccessful Request notifications need to have the View Details link, whereas the successful Request notifications should be plain messages without links

Is there a way to regulate  that?

Reply:
Do we need new parameters in notification_types.yml for supplying more information about the link to the frontend ?

Information such as -
- Whether we need a link `:link: true`  or `:link: false`
- Link text (other areas of the UI may need the text to be something other than "View Details")

Reply:
Theoretically possible, but there are some questions:
* Do you have control over what kind of notification types are getting emitted from automate?
* Are we okay with schema changes?
* Why do we need a custom link text? I am pretty sure it's not a good idea to specify it in the backend ...

Reply:
Or after thinking in the tram, it might be doable through decorators. What if we would have a `notification_link` method into the subject's decorator and move the logic from the [techdebt controller](https://github.com/ManageIQ/manageiq-ui-classic/pull/4505/files#diff-54afc0af5eff6d605e58aea868f5db92R6) to there.

Something like this:
```ruby
class ServiceTemplateTransformationPlanRequestDecorator < MiqDecorator
  def notification_link
    req = ServiceTemplateTransformationPlanRequest.select(:source_id).find(params[:id])
    if aparna_says_link_is_needed(req)
      { :controller => 'migration', :action => 'index', :anchor => "plan/#{req.source_id}" }
    else
      nil
    end
  end
end
```

Generating the text binding:
```ruby
  def text_bindings
    [:initiator, :subject, :cause].each_with_object(text_bindings_dynamic) do |key, result|
      value = public_send(key)
      result[key] = {
        :link => value.decorate.notification_link,
        :text => value.try(:name) || value.try(:description)
      }.compact if value
    end
  end
```

And finally the controller:
```ruby
class RestfulRedirectController < ApplicationController
  def index
    redirect_to(params[:link])
  end
end
```

@AparnaKarve @martinpovolny @himdel what do you think?

Reply:
After thinking this through, maybe it's not the best idea to expose a `redirect_to` so publicly. So it would be much safe to test in the `text_bindings` if the notification has a not`nil` result from the decorator and call the decorator again in the redirect controller...

Reply:
@skateman i like the idea to go with decoartors to determine whether notification link is needed, @AparnaKarve do you agree with this approach?

Reply:
@AparnaKarve the controller was intentionally created by Šimon because it was not possible to do this, at least not universally and until we use explorer screens (which we do).

Reply:
Also I'd need from you the definition of the `aparna_says_link_is_needed` method :wink:

Reply:
@skateman `$window.location.href = path_built_with_controller_action_anchor` seems to work for me, so going via `restful_redirect` seems like a long unnecessary route -- but I leave that implementation up to you.
We can revisit this bit later if necessary.

Also, if we are already in the Migration page, then it would be nice if the  React route `plan\id` kicks in directly (the way it is currently working), that way we don't even have to hit the server.
But again, that could be addressed later.

Reply:
```ruby
def aparna_says_link_is_needed?
  request = ServiceTemplateTransformationPlanRequest.find(params[:id])
  request.request_state == "finished" && request.status != "Ok"
end
```

Reply:
@AparnaKarve thanks :+1:,  yes it would work in your case, but on many other screens it won't and there's a request from @Ladas and @martinpovolny to support clickable links for some provider-related tasks that are accessible only through explorer screens. I would not like to maintain two different implementations until its really necessary.

Reply:
I think that this is getting too complex. The idea with decorators sounds like working around a need to add an attribute.

It's getting too complex.

The notification should be created with all the information needed.

if `request_state == "finished" && status != "Ok"` then set some attribute telling the details are to be displayed. The information if the link should be displayed needs to be part of the notification.

We should not take the notification and do further lookups to decide whether to display details link.

Reply:
@skateman Are we now implementing what I suggested above? https://github.com/ManageIQ/manageiq-ui-classic/pull/4492#discussion_r211814480

https://github.com/ManageIQ/manageiq-schema/pull/263 looks good to me!

Reply:
@AparnaKarve as Martin stated I was trying to avoid changing the schema, but it will be better to do so.